larcoh/dataproc_utils.py

```diff
@@ -0,0 +1,113 @@
+"""
+Utils for submitting the workflow to a Dataproc cluster.
+"""
+import inspect
+import logging
+import math
+import tempfile
+
+from analysis_runner import dataproc
+from cpg_utils import Path
+from cpg_utils.config import get_config
+from cpg_utils.workflows.batch import get_batch
+from hailtop.batch.job import Job
+
+logger = logging.getLogger(__file__)
+
+
+DATAPROC_PACKAGES = [
+    'cpg-utils',
+    'click',
+    'cpg-gnomad==0.6.3',
+    'google',
+    'slackclient',
+    'fsspec',
+    'sklearn',
+    'gcloud',
+    'selenium',
+]
+
+
+# GCP quota on number of cores
+MAX_PRIMARY_WORKERS = 50
+
+
+def dataproc_job(
+    job_name: str,
+    function,
+    function_path_args: dict[str, Path],
+    preemptible: bool = True,
+    num_workers: int | None = None,
+    depends_on: list[Job | None] = None,
+    autoscaling_policy: str | None = None,
+    long: bool = False,
+    worker_boot_disk_size: int | None = None,
+    secondary_worker_boot_disk_size: int | None = None,
+) -> Job:
+    """
+    Submit script as a dataproc job.
+    """
+    script = (
+        f'scripts/dataproc_script.py '
+        f'{function.__module__} {function.__name__} '
+        f'{" ".join([str(p) for p in function_path_args.values()])}'
+    )
+
+    if cluster_id := get_config()['hail'].get('dataproc', {}).get('cluster_id'):
+        # noinspection PyProtectedMember
+        return dataproc._add_submit_job(
+            batch=get_batch(),
+            cluster_id=cluster_id,
+            script=script,
+            pyfiles=['larcoh'],
+            job_name=job_name,
+            region='australia-southeast1',
+        )
+
+    if num_workers is None:
+        num_workers = get_config()['workflow']['scatter_count']
+
+    max_age = '24h' if not long else '48h'
+
+    depends_on = depends_on or []
+    depends_on = [j for j in depends_on if j is not None]
+
+    if autoscaling_policy:
```

Comment:
Should we incorporate this (and other features here) in the generic `hail_dataproc_job` wrapper?